docs/README.md

```diff
@@ -0,0 +1,256 @@
+# Microsoft Azure SDK for Ruby
+
+Microsoft Azure SDK for Ruby has been deprecated. It will be in maintenance mode for a period of time. This doc instructs how to use Azure REST API as a replacement. By the way, it is highly recommanded you to use GA SDKs
+* [Python](https://github.com/Azure/azure-sdk-for-python)
+* [Java](https://github.com/Azure/azure-sdk-for-java)
+* [JavaScript](https://github.com/Azure/azure-sdk-for-js)
+* [.Net](https://github.com/Azure/azure-sdk-for-net)
+
+You can also find other SDKs in [Azure SDK](https://github.com/Azure/azure-sdk).
+
+# Azure REST API
+
+[Azure REST API Reference](https://docs.microsoft.com/en-us/rest/api/azure/)
+
+
+## Authentication
+
+The first step to using the SDK is authentication and permissions. For people unfamilar with Azure this may be one of
+the more difficult concepts. For a reference on setting up a service principal from the command line see
+[Authenticating a service principal with Azure Resource Manager](http://aka.ms/cli-service-principal) or
+[Unattended Authentication](http://aka.ms/auth-unattended). For a more robust explanation of authentication in Azure,
+see [Developer’s guide to auth with Azure Resource Manager API](http://aka.ms/arm-auth-dev-guide).
+
+After creating the service principal, you should have three pieces of information, a client id (GUID), client secret
+(string) and tenant id (GUID) or domain name (string).
+
+# Usage of ms_rest_azure gem
+
+The ms_rest_azure gem is a Azure rest client library used by Azure SDK for Ruby. It relies on faraday and concurrent-ruby. 
+
+## Install
+
+You can install the ms_rest_azure gem with the following command:
+
+```ruby
+gem install 'ms_rest_azure'
+```
+
+## Prerequisite
+
+In order to use the Azure client, you must supply the following values:
+
+* Tenant Id
+* Client Id
+* Subscription Id
+* Client Secret
+
+You could pass the above values in the following ways:
+
+### Option 1 - Environment Variables
+You can set the (above) values using the following environment variables:
+
+* AZURE_TENANT_ID
+* AZURE_CLIENT_ID
+* AZURE_SUBSCRIPTION_ID
+* AZURE_CLIENT_SECRET
+
+To set the environment variables, in Windows, you could use the command such as:
+
+```
+set AZURE_TENANT_ID=<YOUR_TENANT_ID>
+```
+
+In Unix based systems, you could use the command such as:
+
+```
+export AZURE_TENANT_ID=<YOUR_TENANT_ID>
+```
+
+### Option 2 - Options Hash
+The initialization of profile clients take an options hash as a parameter. This options hash consists of tenant_id, client_id, client_secret, subscription_id, active_directory_settings and credentials. Among these, the active_directory_settings and credentials are optional.
```

Comment:
In addition to environment variables, you can also pass those parameters in an Option hash when you initialize the profile clients